pages/filterCards/index.tsx

```diff
@@ -0,0 +1,64 @@
+import axios from "axios";
+// tslint:disable no-submodule-imports
+import { withRouter } from "next/router";
+import * as React from "react";
+
+interface IProps {
+  data: {
+    tags: any;
+  };
+}
+
+class Index extends React.Component<IProps> {
+  public static async getInitialProps() {
+    const res = await axios.get("http://localhost:8080/api/cards/filter");
+    return { data: { tags: res.data.tags } || {} };
+  }
+  public state = { selectedTags: [] as string[] };
+
+  public generateUrl = (ev: React.MouseEvent) => {
+    const target = ev.target as HTMLElement;
+    if (target.tagName === "LABEL") {
+      if (!this.state.selectedTags.some(tag => target.textContent === tag)) {
+        const tag = target.textContent as string;
+        this.state.selectedTags.push(tag);
```

Comment:
DO NOT mutate state https://ru.reactjs.org/tutorial/tutorial.html#why-immutability-is-important

Reply:
Ahh, thanks, thats a huge mistake. It was a night 😴